### DISCUSSION TITLE
How to please

### DISCUSSION THREAD

**Discussion by nadveos** [pocketbase/pocketbase#3696] [Q&A] [answered]
Hi therre, im ve 2 collections. Ones its basically default users and now i ve create post with fileds tittle, content, user_id. im working on flutter and use http package dor initial api call. but response of call only show me id of users collection. Tired trying use ?expand=user_id or something but nothig ocurred. `import 'dart:convert';

import 'package:http/http.dart' as http;
import 'package:pocketbase/pocketbase.dart';
import 'package:portfolio/post_model.dart';

final pocketbase = PocketBase('https://fulltest.xxx.com');

class ApiConnect {
  List<Posts> items = [];
  Future<Posts> getData() async {
    final url = Uri.parse(
        'https://fulltest.xxx.xxx/api/collections/posts/records?expand=user_id');
    final data = await http.get(url);

    Posts post = Posts.fromJson(jsonDecode(data.body));
    
    return post;
    
  }
}`
Otherwise i dont know how to use getList method using pb package with my dto models...ever says me <RecordModel> cant assigned to MyModel. Thamks in advance and Best regards

**Comment by ganigeorgiev** [comment]
I'm having difficulties understanding the description and the provided code sample.

If the `expand` is not being resolved, then double check that the correct relation field name is used and that **the client request satisfies the View API rule of the related collection** (_in your case I guess that would be the "users" collection_).

Here is an example using the PocketBase Dart SDK:

```dart
final pb =  PocketBase('...');

final resultList = await pb.collection('posts').getList(
  page: 1,
  perPage: 20,
  expand: 'user_id',
);
```

**Reply by nadveos** [reply]
![image](https://github.com/pocketbase/pocketbase/assets/94124026/543d699a-b8a6-404a-bd43-6024c83abeea)
![image](https://github.com/pocketbase/pocketbase/assets/94124026/64d5de7c-6470-446a-a84d-9a6f900347cb)
![image](https://github.com/pocketbase/pocketbase/assets/94124026/1dc0bd57-cfe5-4210-beb9-18a81607127f)
sorry for that. my english its so basic. Here is my problem i Try to realationship posts with users but response everytime give me id of user. In olders versions of PB we can select what fields bw show. I need show name of user. Thamnks for answer me.

**Reply by ganigeorgiev** [reply, marked as answer]
>  Try to realationship posts with users but response everytime give me id of user. 

The `expand` parameter will not replace the existing `user_id` value. It will populate its result under the `expand.yourRelField` map key, aka. there must be an `expand` field within each PostItem.

Also I don't understand why you need to send a raw `http.get` request and not use the `getList` of the SDK.

If you want to populate your DTO with the result of the SDK response object, you should be able to call something like:

```dart
final resultList = await pb.collection('posts').getList(
  page: 1,
  perPage: 20,
  expand: 'user_id',
);

Posts post = Posts.fromJson(resultList.toJson());
```

To help you with the debugging, you can also explore the returned raw JSON response by opening your request URL directly in the browser.

**Reply by nadveos** [reply]
thanks so much.
